### ISSUE TITLE
Snapshot retention is deleting more snapshots as defined by policy in case system was pausing snapshots

### ISSUE THREAD

**Issue by goarsna** [kopia/kopia#3758]
Hey there,
I found a strange behavior in the retention rules. What I try is to keep 14 snapshots tagged as daily and 24 snapshots tagged as hourly for my private computer that might be turned of for some time (due to vacation or something like that). When I turn on my computer after it has been shut down for a longer time, I discover the following behavior:

### Bug reproducer
* Set the retention policy to keep 14 daily and 24 hourly snapshots
* Shutdown system
* Power on system after more than 14 days

### Actual result
On the next snapshot run all old snapshots will be deleted.

### Expected result
As set in the retention rules I would expect the last 24 snapshots tagged as "hourly" and the last 14 snapshots tagged as "daily" to be kept.

In other words, it seems like the retention algorithm is evaluating the retention policy rather as "I'll keep snapshots tagged as hourly that have been created in the last 24 hours" than "I'll keep the last 24 snapshots tagged as hourly".

**Comment by okoetter** [comment]
I think this is not a big but working as expected. 
See here: 
https://kopia.discourse.group/t/kopia-snapshot-retention-policies-demystified/2941

> Except for Latest Snapshots, Kopia considers the number you write here as duration, not amount. For example, in this image, 48 is not 48 hourly snapshots, but rather, retain 1 snapshot per hour within 48 hours from your most recent snapshot
